services/clusters-keeper/src/simcore_service_clusters_keeper/modules/ec2.py

```diff
@@ -72,22 +74,27 @@ async def get_ec2_instance_capabilities(
         instance_type_names: set[InstanceTypeType],
     ) -> list[EC2InstanceType]:
         """instance_type_names must be a set of unique values"""
-        instance_types = await self.client.describe_instance_types(
-            InstanceTypes=list(instance_type_names)
-        )
-        list_instances: list[EC2InstanceType] = []
-        for instance in instance_types.get("InstanceTypes", []):
-            with contextlib.suppress(KeyError):
-                list_instances.append(
-                    EC2InstanceType(
-                        name=instance["InstanceType"],
-                        cpus=instance["VCpuInfo"]["DefaultVCpus"],
-                        ram=parse_obj_as(
-                            ByteSize, f"{instance['MemoryInfo']['SizeInMiB']}MiB"
-                        ),
+        try:
+            instance_types = await self.client.describe_instance_types(
+                InstanceTypes=list(instance_type_names)
+            )
+            list_instances: list[EC2InstanceType] = []
+            for instance in instance_types.get("InstanceTypes", []):
+                with contextlib.suppress(KeyError):
+                    list_instances.append(
+                        EC2InstanceType(
+                            name=instance["InstanceType"],
+                            cpus=instance["VCpuInfo"]["DefaultVCpus"],
+                            ram=parse_obj_as(
+                                ByteSize, f"{instance['MemoryInfo']['SizeInMiB']}MiB"
```

Comment:
I see. So why not to use direclty pydantic? this way you do not need to `parse_obj_as` per field

Reply:
I do not need parsing all the time. I use pydantic if parsing is needed, in this case I like the ByteSize type, and it's a shortcut. But I can replace this with just bytes. instead.